src/libtexture/imagecache.cpp

```diff
@@ -467,6 +470,7 @@ ImageCacheFile::open (ImageCachePerThreadInfo *thread_info)
     // Since each subimage can potentially have its own mipmap levels,
     // keep track of the highest level discovered
```

Comment:
No, we're looping over subimages and MIP levels and SUMMING to get the total data size.

For file size, the calculation is outside the loop (see line 586) and just asks for Filesystem::file_size(). Just like there is not a reliable way to associate a size on disk with one particular tile, neither is there for one MIP level or one subimage. It's all or nothing for that stat.